### PR TITLE
feat(async/delay): add persistent option

### PR DESCRIPTION
**Pull request by iuioiua**
For #1730. Am I right in saying a test can't currently be done for this?

### CODE REVIEW

**Comment by 0f-0b** [comment]
`unrefTimer` should probably be added to `_deno_unstable.ts`.

**Comment by kt3k** [comment]
> unrefTimer should probably be added to _deno_unstable.ts.

Right. Let's wrap it with type guard condition `typeof Deno.unrefTimer == "function"`

Also let's mention that the option depends on `--unstable` flag.

**Comment by kt3k** [comment]
Another naming candidate for this option would be `persistent` as used in `fs.watch` of node.js https://nodejs.org/api/fs.html#fswatchfilename-options-listener

Which do you think is better?

File: async/delay.ts [hunk 1]
```diff
@@ -3,11 +3,13 @@
 
 export interface DelayOptions {
   signal?: AbortSignal;
+  /** Defines if the timer should be non-blocking. This is `false` by default. */
```
Comment (kt3k) [inline async/delay.ts:6]: `the timer should be non-blocking` sounds strange to me as timers are always async operations. I would document this like `Indicates whether the process can exit while the timer still existing`

Comment (iuioiua) [reply async/delay.ts:6]: > Another naming candidate for this option would be `persistent` (with opposite effect of boolean value) as used in `fs.watch` of node.js https://nodejs.org/api/fs.html#fswatchfilename-options-listener
> 
> Which do you think is better?

Yeah, I like persistent.

File: async/delay.ts [hunk 2]
```diff
@@ -1,13 +1,16 @@
 // Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
 // This module is browser compatible.
+import { unrefTimer } from "../_deno_unstable.ts";
 
 export interface DelayOptions {
   signal?: AbortSignal;
+  /** Indicates whether the process can exit before the timer ends. This is `false` by default. */
+  persistent?: boolean;
```
Comment (kt3k) [inline async/delay.ts:8]: If we use `persistent`, then the meaning (effect) is the opposite of `unref`. The doc would be `Indicates whether the process should continue to run as long as timer existing` and it defaults to true.

**Review by kt3k — COMMENTED** [review]
@iuioiua Thanks for updating! Left one comment

File: async/delay.ts (hunk 2 above)
Comment (iuioiua) [reply async/delay.ts]: My misunderstanding. Fixed now!

**Review by kt3k — APPROVED** [review]
LGTM, nice!

**Comment by kt3k** [comment]
Let's wait a little while to see if there's any concern on this addition

**Pull request merged**